src/commands/wp.ts

```diff
@@ -0,0 +1,255 @@
+			inputToken
+			sshAuthentication {
+				host
+				port
+				username
+				privateKey
+				passphrase
+			}
+			command {
+				guid
+			}
+		}
+	}
+`;
+
+const getSSHAuthForCommand = async ( appId: number, envId: number, command: string ) => {
+	const api = API();
+
+	return api.mutate< TriggerWPCLICommandMutationResponse >( {
+		mutation: TRIGGER_WP_CLI_COMMAND_MUTATION,
+		variables: {
+			input: {
+				id: appId,
+				environmentId: envId,
+				command,
+			},
+		},
+	} );
+};
+
+export class NonZeroExitCodeError extends Error {
+	public exitCode: number;
+
+	constructor( message: string, exitCode: number ) {
+		super( message );
+		this.exitCode = exitCode;
+	}
+}
+
+export class WPCliCommandOverSSH {
+	private app: App;
+	private env: AppEnvironment;
+	private track: CommandTracker;
+
+	constructor( app: App, env: AppEnvironment ) {
+		this.app = app;
+		this.env = env;
+
+		this.track = makeCommandTracker( 'wp', {
+			app: this.app.id,
+			env: this.env.id,
+			execution_type: 'ssh',
+		} );
+	}
+
+	public async run( command: string ): Promise< void > {
+		if ( ! this.app.id || ! this.env.id ) {
+			await this.track( 'error', {
+				error: 'no_app_env_id',
+				message: 'No app or env ID provided',
+			} );
+
+			throw new Error( 'No app ID or environment ID provided' );
+		}
+
+		debug( "Requesting SSH authentication for command '%s'", command );
+
+		const sshAuth = await getSSHAuthForCommand( this.app.id, this.env.id, command );
+
+		const data = sshAuth.data?.triggerWPCLICommandOnAppEnvironment;
+
+		if ( ! data ) {
+			await this.track( 'error', {
+				error: 'no_ssh_auth_data',
+				message: 'No SSH authentication data received',
+			} );
+
+			throw new Error( 'WP-CLI SSH Authentication failed' );
+		}
+
+		debug( 'Connecting to SSH', {
+			host: data.sshAuthentication.host,
+			port: data.sshAuthentication.port,
+			username: data.sshAuthentication.username,
+			guid: data.command.guid,
+		} );
+
+		try {
+			await this.executeCommandOverSSH( {
+				host: data.sshAuthentication.host,
+				port: data.sshAuthentication.port,
+				username: data.sshAuthentication.username,
+				privateKey: data.sshAuthentication.privateKey,
+				passphrase: data.sshAuthentication.passphrase,
+				guid: data.command.guid,
+				inputToken: data.inputToken,
+			} );
+
+			await this.track( 'success', { guid: data?.command.guid } );
+		} catch ( err ) {
+			if ( err instanceof NonZeroExitCodeError ) {
+				await this.track( 'error', {
+					guid: data?.command.guid,
+					error: 'non_zero_exit_code',
+					message: `Command failed with exit code ${ err.exitCode }`,
+				} );
+
+				process.exit( err.exitCode );
+			}
+			const message = err instanceof Error ? err.message : String( err );
+
+			console.error( `${ chalk.red( 'Error:' ) } ${ message }` );
+
+			await this.track( 'error', {
+				guid: data?.command.guid,
+				error: 'ssh_command_failed',
+				message: 'Error executing command over SSH',
+			} );
+
+			throw new Error( message );
```

Comment:
Instead of throwing, should we just be exiting?

```suggestion
			process.exit( 1 );
```